feat: add `stats/incr/nanmmean`

satansin123 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Resolves #5588.

## Description

> What is the purpose of this pull request?

This pull request:

-  Adds the @stdlib/stats/incr/nanmmean package, which computes an incremental mean while handling NaN values by skipping them.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5588 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

test - 
![image](https://github.com/user-attachments/assets/c5a9bddb-0345-4997-9e3d-94efb7f35d82)

benchmark - 
![image](https://github.com/user-attachments/assets/d5af3a08-a747-4743-b685-894dab60ce10)

examples -
![image](https://github.com/user-attachments/assets/5d62cfe6-28aa-4743-a7c2-c4ec4b925682)


## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

satansin123 committed:
feat: add stats/incr/nanmmean

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented on lib/node_modules/@stdlib/stats/incr/nanmmean/README.md:
```diff
@@ -0,0 +1,184 @@
+
+#### incrnanmmean( window )
+
+Returns an accumulator `function` which incrementally computes a moving [arithmetic mean][arithmetic-mean]. The `window` parameter defines the number of values over which to compute the moving mean.
```
```suggestion
Returns an accumulator function which incrementally computes a moving [arithmetic mean][arithmetic-mean], ignoring `NaN` values.. The `window` parameter defines the number of values over which to compute the moving mean.
```

ShabiShett07 commented on lib/node_modules/@stdlib/stats/incr/nanmmean/README.md:
````diff
@@ -0,0 +1,184 @@
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var randu = require( '@stdlib/random/base/randu' );
+var incrnanmmean = require( '@stdlib/stats/incr/nanmmean' );
+
+var accumulator;
+var v;
+var i;
+
+// Initialize an accumulator:
+accumulator = incrnanmmean( 5 );
+
+// For each simulated datum, update the moving mean...
+for ( i = 0; i < 100; i++ ) {
+    if ( randu() < 0.2 ) {
+        v = NaN;
+    } else {
+        v = randu() * 100.0;
+    }
+    accumulator( v );
````
```suggestion
var uniform = require( '@stdlib/random/base/uniform' );
var bernoulli = require( '@stdlib/random/base/bernoulli' );
var incrnanmmean = require( '@stdlib/stats/incr/nanmmean' );

// Initialize an accumulator:
var accumulator = incrnanmmean( 5 );

// For each simulated datum, update the moving mean...
var I;
for ( i = 0; i < 100; i++ ) {
    accumulator( ( bernoulli( 0.8 ) < 1 ) ? NaN : uniform( 0.0, 100.0 ) );
```

ShabiShett07 commented on lib/node_modules/@stdlib/stats/incr/nanmmean/benchmark/benchmark.js:
```diff
@@ -0,0 +1,69 @@
+// MODULES //
+
+var bench = require( '@stdlib/bench' );
+var randu = require( '@stdlib/random/base/randu' );
```
```suggestion
```

ShabiShett07 replied:
We will be using iterations as the input

ShabiShett07 commented on lib/node_modules/@stdlib/stats/incr/nanmmean/benchmark/benchmark.js:
```diff
@@ -0,0 +1,69 @@
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		v = acc( randu() );
```
```suggestion
		v = acc( i );
```

ShabiShett07 commented on lib/node_modules/@stdlib/stats/incr/nanmmean/examples/index.js:
```diff
@@ -0,0 +1,42 @@
+
+'use strict';
+
+var randu = require( '@stdlib/random/base/randu' );
```
same as mentioned in the markdown file

ShabiShett07 commented on lib/node_modules/@stdlib/stats/incr/nanmmean/test/test.js:
```diff
@@ -0,0 +1,111 @@
+// MODULES //
+
+var tape = require( 'tape' );
+var incrnanmmean = require( '@stdlib/stats/incr/nanmmean/lib' );
```
```suggestion
var incrnanmmean = require( './../lib' );
```

ShabiShett07 commented:
@satansin123, the PR looks good, left some comments to make some changes

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmmean/lib/main.js:
```diff
@@ -0,0 +1,122 @@
+			return mu;
+		}
+
+		if ( isnan( x ) ) {
+			return mu;
+		}
```
When `NaN` is provided before any non-NaN data (`N === 0`), this returns `mu` which is `0.0`. Should return `null` to match no-data semantics. E.g., `incrnanmmean(3)(NaN)` returns `0` but should return `null`.

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmmean/lib/main.js:
```diff
@@ -0,0 +1,122 @@
+
+// MODULES //
+
+var isPositiveInteger = require( '@stdlib/assert/is-positive-integer' ).isPrimitive;
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var Float64Array = require( '@stdlib/array/float64' );
+var format = require( '@stdlib/string/format' );
+
```
All other `nan*` moving accumulators (`nanmsum`, `nanmstdev`, `nanmcv`, etc.) wrap the corresponding base accumulator and filter `NaN`. Please refactor to follow that pattern rather than reimplementing the moving mean from scratch. See `nanmsum/lib/main.js` for reference.

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmmean/docs/repl.txt:
```diff
@@ -0,0 +1,47 @@
+
+{{alias}}( W )
+    Returns an accumulator function which incrementally computes a moving
+    arithmetic mean, ignorning `NaN` values.
```
Typo.

```suggestion
    arithmetic mean, ignoring `NaN` values.
```

Planeshifter committed:
fix: address review feedback

- Refactor to wrap incrmmean instead of reimplementing
- Fix NaN-before-data bug (was returning 0.0, now returns null)
- Fix typo in repl.txt (ignorning -> ignoring)
- Add NaN handling mention in README description
- Replace randu with uniform/bernoulli in README and examples
- Remove randu import from benchmark, use loop index
- Fix format specifiers in examples/index.js
- Use relative require path in test.js
- Remove backticks from NaN in package.json description
- Update copyright years to 2026

Planeshifter commented:
/stdlib merge

Planeshifter reviewed (approved):
Should be fine to land.
